app/components/RangeSlider.tsx

```diff
@@ -0,0 +1,115 @@
+import React, { useState } from 'react';
+import tinygradient from 'tinygradient';
+
+interface RangeSliderProps {
+    min: number;
+    max: number;
+    step: number;
+    initialValue?: number;
+    onChange: (value: number) => void;
+}
+
+const RangeSlider: React.FC<RangeSliderProps> = ({ min, max, step, initialValue = min, onChange }) => {
+    const [value, setValue] = useState<number>(initialValue);
+
+    const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+        const newValue = Number(event.target.value);
+        setValue(newValue);
+        onChange(newValue);
+    };
+
+    const percentage = ((value - min) / (max - min)) * 100;
+
+    // gradient for progress track
+    const gradient = tinygradient("#90E568", "#FFDC6A", "#FD6376");
+    const currentColor = gradient.rgbAt(percentage / 100).toHexString();
```

Comment:
if we're using the tinygradient library we should use HSV interpolation to make full use of its features (and a nicer gradient)